### PR TITLE
gh-99308: Clarify re docs for byte pattern group names

### PR DESCRIPTION
**Pull request by Kentzo**
* Issue: gh-99308

### CODE REVIEW

File: Doc/library/re.rst [hunk 1]
```diff
@@ -496,6 +496,8 @@ The special characters are:
 
    .. versionchanged:: 3.12
       Group *id* can only contain ASCII digits.
+      In bytes replacement strings group names must contain only characters
```
Comment (merwok) [inline Doc/library/re.rst]: Let’s avoid five nouns in a row 😄 
```suggestion
      In bytes replacement strings, group names must contain only characters
```

File: Doc/library/re.rst [hunk 2]
```diff
@@ -496,6 +496,8 @@ The special characters are:
...
+      in the ASCII range.
```
Comment (CAM-Gerlach) [inline Doc/library/re.rst]: Maybe its just me, but this was pretty confusing to me and hard to parse and understand. The comma @merwok suggest helps a lot in making the basic grammar more clear, but there still seems to be some problems with the actual semantics here.

As far as I can tell, this should refer to _patterns_, not _replacements_, and I was very confused why the latter were mentioned until I realized it was incorrectly copy/pasted from the original mention (that talks about replacement strings).

Also, "bytes strings" is both arguably an oxymoron in Python 3 (since this is presumably intended to specifically refer to bytes, not strings), seemingly ungrammatical (at least without being proper type references), and certainly confusing.

Finally, *name* should be formatted consistently with *id* in the previous line, :class:`bytes` objects are composed of bytes, not characters (though perhaps that's being too pedantic), and it would be useful to explictly specify what the ASCII range is. Implementing all of that, and fixing the phrasing, we have:

```suggestion
      In :class:`bytes` patterns, group *name* can only contain bytes
      in the ASCII range (``b'\x00'``-``b'\x7f'``).
```

Or, with decimal values for the ASCII range,

```suggestion
      In :class:`bytes` patterns, group *name* can only contain bytes
      in the ASCII range (values 0-127).
```

Comment (Kentzo) [reply Doc/library/re.rst]: @CAM-Gerlach Maybe `b'\x00'`-`b'\x7f'`?

Comment (CAM-Gerlach) [reply Doc/library/re.rst]: I guess so; edited. I just didn't want to overcomplicate it.

**Review by CAM-Gerlach — APPROVED** [review]
Seems LGTM now from my side, thanks. :+1:

**Review by hauntsaninja — APPROVED** [review]
Thanks for improving this!

This looks good to me, although my ignorable nit is that "ASCII range" alone seems sufficient to me: specifying the hex values doesn't add much since group names need to be valid Python identifiers.

**Review by kumaraditya303 — APPROVED** [review]

**Comment by kumaraditya303** [comment]
Skipping backport as it is for 3.12 only.
